controllers/configmap_controller.go

```diff
@@ -168,6 +168,29 @@ func validateAddress(address string) error {
 	return nil
 }
 
+// getNodeUsername retrieves the username associated with the given node from the instance ConfigMap data
+func getNodeUsername(configMapData map[string]string, node *core.Node) (string, error) {
+	if node == nil {
+		return "", errors.New("cannot get username for nil node")
+	}
+	// Find entry in ConfigMap that is associated to node via address
+	for _, address := range node.Status.Addresses {
+		if data, found := configMapData[address.Address]; found {
+			return extractUsernameFromEntry(data)
+		}
+	}
+	return "", errors.Errorf("unable to find instance associated with node %s", node.GetName())
+}
+
+// extractUsernameFromEntry returns the username string from data in the form username=<username>
```

Comment:
What's an `Entry`? I feel `extractUsername` as function name is good.

Reply:
+1